dimension/id_cache.go

```diff
@@ -0,0 +1,47 @@
+package dimension
+
+import (
+	"github.com/patrickmn/go-cache"
+	"time"
+)
+
+// MemoryCache is an in memory cache of dimensions with database id's.
+type MemoryCache struct {
+	store       IDStore
+	memoryCache *cache.Cache // instanceID > dimensionName > nodeId
+}
+
+// IDStore represents the data store for dimension id's
+type IDStore interface {
+	GetIDs(instanceID string) (map[string]string, error)
+}
+
+// NewIDCache returns a new cache instance that uses the given data store.
+func NewIDCache(idStore IDStore, cacheTTL time.Duration) *MemoryCache {
+	return &MemoryCache{
+		store:       idStore,
+		memoryCache: cache.New(cacheTTL, 15*time.Minute),
+	}
+}
+
+// GetNodeIDs returns all dimensions for a given instanceID
+func (mc *MemoryCache) GetNodeIDs(instanceID string) (map[string]string, error) {
+
+	dimensions, ok := mc.memoryCache.Get(instanceID)
+
+	if ok {
+		return dimensions.(map[string]string), nil
+	}
+
+	newDimensions, err := mc.store.GetIDs(instanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = mc.memoryCache.Add(instanceID, newDimensions, cache.DefaultExpiration)
+	if err != nil {
+		return map[string]string{}, err
```

Comment:
Seems reasonable to. I will make the change and ask Matt if there is a reason to change it back.